### PR TITLE
1578: Do not reload date range report

### PR DESCRIPTION
**Pull request by kael89**
### Issue #:
Fix https://github.com/beyondessential/tupaia-backlog/issues/1578

### CODE REVIEW

File: packages/web-config-server/src/apiV1/view.js [hunk 1]
```diff
@@ -1,10 +1,13 @@
-import { convertDateRangeToPeriods, replaceValues, CustomError } from '@tupaia/utils';
+import { convertDateRangeToPeriods, CustomError, replaceValues, utcMoment } from '@tupaia/utils';
 import { getDhisApiInstance } from '/dhis';
 import { isSingleValue } from './utils';
 import { DataAggregatingRouteHandler } from './DataAggregatingRouteHandler';
 import { DashboardPermissionsChecker } from './permissions';
 import { getDataBuilder } from '/apiV1/dataBuilders/getDataBuilder';
 
+// Same as in @tupaia/web-frontend
+const DEFAULT_MIN_DATE = '20150101';
```
Comment (kael89) [inline packages/web-config-server/src/apiV1/view.js:9]: This is also defined in `web-fronted/src/components/DateRangePicker/constants.js`.

Ideally we would define this constant in a single place (eg the `utils` package), but we've had a bit of trouble loading functionality from other tupaia packages in `web-frontend`. Hopefully the above comment helps!

File: packages/web-config-server/src/apiV1/view.js [hunk 2]
```diff
@@ -113,7 +109,12 @@ export default class extends DataAggregatingRouteHandler {
 
   // common view translation (for all possible views)
   addViewMetaData = inJson => {
-    const { viewJson, query, startDate, endDate } = this;
+    const {
+      viewJson,
+      query,
+      startDate = DEFAULT_MIN_DATE,
+      endDate = utcMoment().format('YYYYMMDD'),
```
Comment (kael89) [inline packages/web-config-server/src/apiV1/view.js:116]: The front-end uses these values as defaults for start and end dates, so we may as well as specify them earlier on in the data flow. Ideally we would infer them from the results, but it's hard to do so for all data builders given the existing architecture. The new `reports` package will probably make this easier!

**Review by chris-bes — APPROVED** [review]
LGTM!

**Comment by kael89** [comment]
@chris-bes after consulting @edmofro I realised that there is no need to have the default date range logic in `web-config-server`, but we should move it in the front end saga level. Could you please re-review ?

File: packages/web-frontend/src/actions.js [hunk 3]
```diff
@@ -697,17 +697,13 @@ export function fetchDashboardItemData(
   dashboardGroupId,
   viewId,
   infoViewKey,
-  startDate,
-  endDate,
```
Comment (kael89) [inline packages/web-frontend/src/actions.js:701]: Those were never used in practice

**Review by chris-bes — APPROVED** [review]

**Comment by kael89** [comment]
@Alastair-L this is "ready for test" now, just bringing it in your attention in case it conflicts with https://github.com/beyondessential/tupaia-backlog/issues/1421

**Comment by Alastair-L** [comment]
Thanks @kael89 for bringing it to my attention!

**Pull request merged**